lib/ransack/search.rb

```diff
@@ -27,11 +27,12 @@ def result(opts = {})
 
     def build(params)
       collapse_multiparameter_attributes!(params).each do |key, value|
-        case key
-        when 's', 'sorts'
+        if key == 's' || key == 'sorts'
           send("#{key}=", value)
-        else
-          base.send("#{key}=", value) if base.attribute_method?(key)
+        elsif @base.attribute_method?(key)
+          base.send("#{key}=", value)
+        elsif @context.ransackable_scope?(key, @context.object)
```

Comment:
That might be workable, I think I suggested [something like it][0] but it was just easier to build into the existing structure, see my [original comment][1]. Part of it was so things like ActiveAdmin and RansackUI would "just work" but maybe not the cleanest approach.

[0]: https://github.com/activerecord-hackery/ransack/pull/288#issuecomment-26762566
[1]: https://github.com/activerecord-hackery/ransack/pull/288#issuecomment-28419062